lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -243,6 +244,41 @@ func (c *ContainerProxy) CreateContainerHandle(vc *viccontainer.VicContainer, co
 	return id, h, nil
 }
 
+// AddImageToContainer adds the specified image to a container, referenced by handle.
+// If an error is return, the returned handle should not be used.
+//
+// returns:
+//	modified handle
+func (c *ContainerProxy) AddImageToContainer(handle, deltaID, layerID, imageID string, config types.ContainerCreateConfig) (string, error) {
+	defer trace.End(trace.Begin(handle))
+
+	if c.client == nil {
+		return "", InternalServerError("ContainerProxy.AddImageToContainer failed to get the portlayer client")
+	}
+
+	host, err := sys.UUID()
+	if err != nil {
+		return "", InternalServerError("ContainerProxy.AddImageToContainer got unexpected error getting VCH UUID")
+	}
+
+	response, err := c.client.Storage.ImageJoin(storage.NewImageJoinParamsWithContext(ctx).WithStoreName(host).WithID(layerID).
```

Comment:
This reads funny.  We're joining an image to a layer?  Should WithID() be given the deltaID, instead?

Reply:
We're joining an image from a given image store and identified by it's ID.
The ID of the image to join is held in layerID (the leaf of the image tree).
The ID to use for creating a read/write layer based on the image we're joining is in deltaID.

Basically we've got lots of IDs so there's going to be some nomenclature confusion